app/src/main/java/org/oppia/app/player/state/StateViewModel.kt

```diff
@@ -49,7 +51,15 @@ class StateViewModel @Inject constructor() : ObservableViewModel() {
     return getPendingAnswerHandler(itemList)?.isExplicitAnswerSubmissionRequired() ?: true
   }
 
-  // TODO(#164): Add a hasPendingAnswer() that binds to the enabled state of the Submit button.
+  /** Returns whether there is currently a pending interaction that also acts like a navigation button. */
+  fun isMostRecentInteractionAutoNavigating(itemList: List<StateItemViewModel>): Boolean {
+    return getPendingAnswerHandler(itemList)?.isAutoNavigating() ?: false
+  }
+
+  fun setCanSubmitAnswer(canSubmitAnswer: Boolean) = this.canSubmitAnswer.set(canSubmitAnswer)
+
+  fun getCanSubmitAnswer(): ObservableField<Boolean> = canSubmitAnswer
+
```

Comment:
@veena14cs There is one function in this file `setHintRevealedVisibility` which is not getting used. Are you using it somewhere?

Reply:
This is not used now, it was added to show radar effect when a new hint is available but its unrevealed.

Reply:
Thanks for the confirmation, Veena!